### PR TITLE
Fix decompression functions not returning errcodes

### LINKED ISSUE #57049: Corrupted compressed file can be read without errors in GDscript
### Godot version

3.4.2

### System information

Windows 11

### Issue description

When writing a compressed file with Godot and then intentionally modifying it using a hex editor, the file can still be opened by Godot and does not return an error in GDscript (`file.open_compressed(OUTPUT_FILE, File.READ, COMPRESSION)` returns `OK`). It still prints an error though (can't be detected AFAIK in GDscript). When the corupted part of the file is farer away from the beginning of the file (i presume because it does not fit into the intial buffer), the error is printed as well when calling one of the `get_*` methods. But `get_error()` is always `OK`. 

This is the error:
```
E 0:00:20.310   decompress: Condition "err != 1" is true. Returned: -1
  <C++ Source>  core/io/compression.cpp:166 @ decompress()
  <Stack Trace> Main.gd:22 @ _on_ReadButton_pressed()
```

The return value of decompress method is dicarded in all instances when it is called in `file_access_compressed.cpp`. For example:
https://github.com/godotengine/godot/blob/45eaa2daf1e3d2571a9f31a3421911262ec82f51/core/io/file_access_compressed.cpp#L272-L278

Shouldn't this return `ERR_FILE_CORRUPT` when `decompress()` returns -1 so that the user knows the file is corrupt?

### Steps to reproduce

1. Write a compressed file (compression method does not matter i think) using Godot

````gdscript
var file := File.new()
var res := file.open_compressed(OUTPUT_FILE, File.WRITE, File.COMPRESSION_GZIP)
if res != OK:
	printerr("Failed to open file for writing")
	return

file.store_string("Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua")
file.close()
print("File written!")
````

2. Open the file in a Hex Editor and change something (not the magic number at the start)
3. Read the file using Godot again

```gdscript
var file := File.new()
var res := file.open_compressed(OUTPUT_FILE, File.READ, File.COMPRESSION_GZIP)
if res != OK:
	printerr("Failed to open file for reading")
	return
print(file.get_error())
	
file.get_buffer(file.get_len())
print(file.get_error())
file.close()
```
`file.get_error()` and `file.open_compressed()` will not report errors. But you can see some errors in the Debugger.

### Minimal reproduction project

[FileCompressionBug.zip](https://github.com/godotengine/godot/files/7916651/FileCompressionBug.zip)

### PR DESCRIPTION
**Pull request by rsubtil**
`Compression::decompress` returns -1 on error scenarios. This was not being propagated to let the user know there was an error while reading compressed files.

Fixes https://github.com/godotengine/godot/issues/57049

### CODE REVIEW

File: core/io/file_access_compressed.cpp [hunk 1]
```diff
@@ -305,7 +308,8 @@ uint64_t FileAccessCompressed::get_buffer(uint8_t *p_dst, uint64_t p_length) con
 			if (read_block < read_block_count) {
 				//read another block of compressed data
 				f->get_buffer(comp_buffer.ptrw(), read_blocks[read_block].csize);
-				Compression::decompress(buffer.ptrw(), read_blocks.size() == 1 ? read_total : block_size, comp_buffer.ptr(), read_blocks[read_block].csize, cmode);
+				int err = Compression::decompress(buffer.ptrw(), read_blocks.size() == 1 ? read_total : block_size, comp_buffer.ptr(), read_blocks[read_block].csize, cmode);
```
Comment (akien-mga) [inline core/io/file_access_compressed.cpp:311]: Should be `ret` for consistency with other checks.

File: core/io/file_access_compressed.cpp [hunk 2]
```diff
@@ -125,10 +125,11 @@ Error FileAccessCompressed::_open(const String &p_path, int p_mode_flags) {
 		char rmagic[5];
 		f->get_buffer((uint8_t *)rmagic, 4);
 		rmagic[4] = 0;
-		if (magic != rmagic || open_after_magic(f) != OK) {
+		Error err = ERR_FILE_UNRECOGNIZED;
```
Comment (akien-mga) [inline core/io/file_access_compressed.cpp:128]: Shadows previous `err`:
```suggestion
		err = ERR_FILE_UNRECOGNIZED;
```

File: core/io/file_access_compressed.cpp [hunk 3]
```diff
@@ -273,7 +275,8 @@ uint8_t FileAccessCompressed::get_8() const {
 		if (read_block < read_block_count) {
 			//read another block of compressed data
 			f->get_buffer(comp_buffer.ptrw(), read_blocks[read_block].csize);
-			Compression::decompress(buffer.ptrw(), read_blocks.size() == 1 ? read_total : block_size, comp_buffer.ptr(), read_blocks[read_block].csize, cmode);
+			int ret = Compression::decompress(buffer.ptrw(), read_blocks.size() == 1 ? read_total : block_size, comp_buffer.ptr(), read_blocks[read_block].csize, cmode);
+			ERR_FAIL_COND_V_MSG(ret == -1, 0, "Compressed file is corrupt.");
```
Comment (akien-mga) [inline core/io/file_access_compressed.cpp:279]: This shadows `ret` which is actually the decompression result.

File: core/io/file_access_compressed.cpp (hunk 1 above)
Comment (Ev1lbl0w) [reply core/io/file_access_compressed.cpp]: Fixed :+1:

File: core/io/file_access_compressed.cpp (hunk 2 above)
Comment (Ev1lbl0w) [reply core/io/file_access_compressed.cpp]: Fixed :+1:

File: core/io/file_access_compressed.cpp (hunk 3 above)
Comment (Ev1lbl0w) [reply core/io/file_access_compressed.cpp]: Fixed, I picked `total` as a more sensible name than `err` for this one.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 3.5.

**Comment by akien-mga** [comment]
Cherry-picked for 3.4.3.

**Pull request merged**